### PR TITLE
x86disp: Fix ROL/ROR CF and OF calculation

### PR DESCRIPTION
**Pull request by skmp**

### CODE REVIEW

File: unittests/ASM/PrimaryGroup/2_D3_00_2.asm [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+mov rax, 0x0001
+
+mov cl, 16
+rol ax, cl
+
+mov r13, 0
+cmovc r13, r15
+
+; 32bit
+mov rax, 0x00000020
```
Comment (skmp) [inline unittests/ASM/PrimaryGroup/2_D3_00_2.asm:38]: @Sonicadvance1 should this be 0x1 here?

File: unittests/ASM/PrimaryGroup/2_D3_00_2.asm [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+mov cl, 31
+rol eax, cl
+
+mov r12, 0
+cmovc r12, r15
+
+; 64bit
+mov rax, 0x0000000000000020
```
Comment (skmp) [inline unittests/ASM/PrimaryGroup/2_D3_00_2.asm:47]: @Sonicadvance1 should this be 0x1 here?

File: unittests/ASM/PrimaryGroup/2_D3_00_3.asm [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+rol rax, cl
+
+mov r11, 0
+cmovo r11, r15
+
+; Let's clear OF really quick
+mov rax, 0
+rol rax, 1
+
+; Shouldn't set OF
```
Comment (skmp) [inline unittests/ASM/PrimaryGroup/2_D3_00_3.asm:59]: @Sonicadvance1 Won't these set OF as OF = MSB(result)^CF and CF=1 as the top bit rotates?

File: unittests/ASM/PrimaryGroup/2_D3_01_3.asm [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+ror rax, cl
+
+mov r11, 0
+cmovo r11, r15
+
+; Let's clear OF really quick
+mov rax, 0
+ror rax, 1
+
+; Shouldn't set OF
```
Comment (skmp) [inline unittests/ASM/PrimaryGroup/2_D3_01_3.asm:59]: won't these set OF as OF = DEST[MSB-1] ^ CF and CF = 0?

File: unittests/ASM/PrimaryGroup/2_D3_00_3.asm (hunk 3 above)
Comment (Sonicadvance1) [reply unittests/ASM/PrimaryGroup/2_D3_00_3.asm:59]: OF isn't set here since flags aren't updated when shift is zero

File: unittests/ASM/PrimaryGroup/2_D3_01_3.asm (hunk 4 above)
Comment (Sonicadvance1) [reply unittests/ASM/PrimaryGroup/2_D3_01_3.asm:59]: OF isn't set here since flags aren't updated when shift is zero

File: unittests/ASM/PrimaryGroup/2_D3_00_2.asm (hunk 2 above)
Comment (Sonicadvance1) [reply unittests/ASM/PrimaryGroup/2_D3_00_2.asm:47]: Actually was meant to be 0x02

File: unittests/ASM/PrimaryGroup/2_D3_00_2.asm (hunk 1 above)
Comment (Sonicadvance1) [reply unittests/ASM/PrimaryGroup/2_D3_00_2.asm:38]: Actually was meant to be 0x02

File: External/FEXCore/Source/Interface/Core/OpcodeDispatcher.cpp [hunk 5]
```diff
@@ -1712,6 +1712,10 @@ void OpDispatchBuilder::ASHROp(OpcodeArgs) {
   else
     Src = _And(Src, _Constant(Size, 0x1F));
 
+  if (Size != 64) {
+    Dest = _Sbfe(Size, 0, Dest);
```
Comment (Sonicadvance1) [inline External/FEXCore/Source/Interface/Core/OpcodeDispatcher.cpp:1716]: Just need this unrelated fix yeeted out of here and it should be good.

**Review by Sonicadvance1 — APPROVED** [review]
